8267708: Remove references to com.sun.tools.javadoc.**

pavelrappo opened the pull request:
Although the `com.sun.tools.javadoc.**` packages were removed in JDK-8215584, some references to them remain, mainly in the form of links in doc comments. Some of those links refer to classes that never existed, which I reckon is due to a typo; for example, this reference

    com.sun.tools.javadoc.ClassDocImpl#findField

should've looked like this

    com.sun.tools.javadoc.main.ClassDocImpl#findField

Grep showed that some tests also use `com.sun.tools.javadoc` and some other extinct packages. However, it is unclear if those "references" should also be removed:

1. https://github.com/openjdk/jdk/blob/739769c8fc4b496f08a92225a12d07414537b6c0/test/langtools/tools/javac/NoStringToLower.java#L60-L74
2. https://github.com/openjdk/jdk/blob/739769c8fc4b496f08a92225a12d07414537b6c0/test/langtools/tools/javac/T8003967/DetectMutableStaticFields.java#L73-L85

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8267708](https://bugs.openjdk.java.net/browse/JDK-8267708): Remove references to com.sun.tools.javadoc.**


### Reviewers
 * [Jonathan Gibbons](https://openjdk.java.net/census#jjg) (@jonathan-gibbons - **Reviewer**)


### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4192/head:pull/4192` \
`$ git checkout pull/4192`

Update a local copy of the PR: \
`$ git checkout pull/4192` \
`$ git pull https://git.openjdk.java.net/jdk pull/4192/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4192`

View PR using the GUI difftool: \
`$ git pr show -t 4192`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4192.diff">https://git.openjdk.java.net/jdk/pull/4192.diff</a>

</details>

jonathan-gibbons commented on src/jdk.compiler/share/classes/com/sun/tools/javac/api/JavacTrees.java:
```diff
@@ -624,12 +624,10 @@ private Symbol attributeParamIdentifier(TreePath path, DCParam paramTag) {
         return null;
     }
 
-    /** @see com.sun.tools.javadoc.ClassDocImpl#findField */
```
Is there an equivalent new-style method that should be used instead?

jonathan-gibbons replied:
The most likely justification for all these ClassDocImpl references is now in `ReferenceTree`.

pavelrappo replied:
A doc comment consisting solely of an `@see` tag has unclear semantics. The methods I removed such doc comments from are private and can be used only from their enclosing instance.

jonathan-gibbons replied:
Let me go do some archaeology. While the comments may have been substandard, I would still like to recall their intent.

jonathan-gibbons replied:
The notable content of the comments on those `ClassDocImpl` methods is that they document (I hesitate to use the "specify" word) the search order, with the additional comment *Note that this is not necessarily what the compiler would do!*

jonathan-gibbons commented on src/jdk.compiler/share/classes/com/sun/tools/javac/util/Log.java:
```diff
@@ -312,11 +312,6 @@ protected Log(Context context, PrintWriter out, PrintWriter err) {
 
     /**
      * Construct a log with given I/O redirections.
-     * @deprecated
-     * This constructor is provided to support the supported but now-deprecated javadoc entry point
-     *      com.sun.tools.javadoc.Main.execute(String programName,
-     *          PrintWriter errWriter, PrintWriter warnWriter, PrintWriter noticeWriter,
-     *          String defaultDocletClassName, String... args)
```
It is generally bad form to have `@Deprecated` without `@deprecated`. In this case, separate ongoing work to cleanup `Messager` will eventually mean we can remove the method altogether.

If you really want to remove the reference to the old code, the `@deprecated` comment should be updated to "Still in use by javadoc's Mesager class." or some such.

jonathan-gibbons reviewed (changes requested):
The package references in the two tests you found should be updated to use `jdk.javadoc`.  That's already present for #2, so just delete the old package in #2, and change the package in #1.

pavelrappo commented:
1. Removed extinct packages
2. Removed `ignore("jdk/javadoc/internal/tool/Start", "versionRB")` since `Start.versionRB` was removed in JDK-8246078
3. Returned updated deprecation info.

pavelrappo commented:
When I fixed the location of javadoc, the NoStringToLower test found two issues it could not see before, both of which I then fixed.

I also note, that this method in DetectMutableStaticFields should perhaps use `String.startsWith` instead of `String.contains` in which case `packagesToSeekFor` should be also renamed to `packagePrefixesToSeekFor`:

    boolean shouldAnalyzePackage(String packageName) {
        for (String aPackage: packagesToSeekFor) {
            if (packageName.contains(aPackage)) {
                return true;
            }
        }
        return false;
    }

But if this is to be fixed, it should be done in a separate PR.

jonathan-gibbons reviewed (approved):
See the updated review comments about removing the obsolete doc comments.

pavelrappo commented:
/summary This commit changes BaseOptions.java to fix the NoStringToLower test that failed after jdk.javadoc has been added to the list of package prefixes that NoStringToLower scans.

pavelrappo commented:
/integrate
